### PR TITLE
Fix ReadPacket and WritePacket payload length is a multiple

### PR DESCRIPTION
**Pull request by michaelyou**
MaxPayloadLen

#208

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @WangXiangUSTC

**Comment by WangXiangUSTC** [comment]
@michaelyou could you please tell me how to reappear the bug?  provide a test case

**Comment by michaelyou** [comment]
I think it's obvious. If the payload len is a multiply of MaxPayloadLen, the origin code will read once more, but there is nothing for reading. you will get an error in line 81.

**Comment by michaelyou** [comment]
This bug may not happen in practical use(the probability is really small that the payload has a length of exactly MaxLengthlen)，but it really exists logically.

**Comment by siddontang** [comment]
I think we should add a test now. Conn needs a net.Conn which we can construct a dummy one in the test with the length of payload equal MaxLength

**Comment by WangXiangUSTC** [comment]
@michaelyou can you help us add a uint test of this function? otherwise it is difficult to check  the correctness

**Comment by danieltahara** [comment]
Is this going to be merged? seems like a test like https://github.com/vitessio/vitess/pull/3935/files#diff-6a379d5bc6a249add52fb846b2ee9c3f would be sufficient

File: packet/conn.go [hunk 1]
```diff
@@ -30,124 +29,87 @@ func NewConn(conn net.Conn) *Conn {
...
+		if _, err := io.ReadFull(c.br, header); err != nil {
+			return nil, ErrBadConn
+		}
 
-	length := int(uint32(header[0]) | uint32(header[1])<<8 | uint32(header[2])<<16)
-	if length < 1 {
-		return errors.Errorf("invalid payload length %d", length)
-	}
+		// packet length [24 bit]
+		length := int(uint32(header[0]) | uint32(header[1])<<8 | uint32(header[2])<<16)
```
Comment (GregoryIan) [inline packet/conn.go]: Why not checking length with max payload length here?

**Comment by lance6716** [comment]
closed by https://github.com/go-mysql-org/go-mysql/issues/208#issuecomment-2490173102

**Pull request closed without merging**